mantle/kola/tests/misc/raid.go

```diff
@@ -173,16 +176,31 @@ func checkIfMountpointIsRaid(c cluster.TestCluster, m platform.Machine, mountpoi
 // is found to be mounted at /.
 func checkIfMountpointIsRaidWalker(c cluster.TestCluster, bs []blockdevice, mountpoint string) bool {
 	for _, b := range bs {
-		if b.Mountpoint != nil && *b.Mountpoint == mountpoint {
+		if checkIfBlockdevHasMountPoint(b, mountpoint) {
 			if b.Type != "raid1" {
 				c.Fatalf("device %q is mounted at %q with type %q (was expecting raid1)", b.Name, mountpoint, b.Type)
 			}
 			return true
 		}
-		foundRoot := checkIfMountpointIsRaidWalker(c, b.Children, mountpoint)
-		if foundRoot {
+		foundDevice := checkIfMountpointIsRaidWalker(c, b.Children, mountpoint)
+		if foundDevice {
 			return true
 		}
 	}
 	return false
 }
+
+// checkIfBlockdevHasMountPoint checks if a given block device has the
+// required mountpoint.
+func checkIfBlockdevHasMountPoint(b blockdevice, mountpoint string) bool {
+	if b.Mountpoint != nil && *b.Mountpoint == mountpoint {
+		return true
+	} else if len(b.Mountpoints) != 0 {
```

Comment:
Minor/optional: could drop this and just always go to the for loop since it'll handle empty slices fine.